server/command.go

```diff
@@ -0,0 +1,345 @@
+	if len(args) != 1 {
+		return responsef(helpText)
+	}
+	return executeEnableImpl(p, c, header, args[0], true)
+}
+
+func executeDisable(p *Plugin, c *plugin.Context, header *model.CommandArgs, args ...string) *model.CommandResponse {
+	if len(args) != 1 {
+		return responsef(helpText)
+	}
+	return executeEnableImpl(p, c, header, args[0], false)
+}
+
+func executeEnableImpl(p *Plugin, c *plugin.Context, header *model.CommandArgs, ref string, enabled bool) *model.CommandResponse {
+	links, refs, err := parseLinkRef(p, true, ref)
+	if err != nil {
+		return responsef("%v", err)
+	}
+	l := &links[refs[0]]
+	l.Disabled = !enabled
+
+	err = saveConfigLinks(p, links)
+	if err != nil {
+		return responsef(err.Error())
+	}
+
+	if l.Name != "" {
+		ref = l.Name
+	}
+	return executeList(p, c, header, ref)
+}
+
+func executeAdd(p *Plugin, c *plugin.Context, header *model.CommandArgs, args ...string) *model.CommandResponse {
+	if len(args) > 1 {
+		return responsef(helpText)
+	}
+	name := ""
+	if len(args) == 1 {
+		name = args[0]
+	}
+
+	err := saveConfigLinks(p, append(p.getConfig().Links, Link{
+		Name: name,
+	}))
+	if err != nil {
+		return responsef(err.Error())
+	}
+
+	if name == "" {
+		return executeList(p, c, header)
+	}
+	return executeList(p, c, header, name)
+}
+
+func responsef(format string, args ...interface{}) *model.CommandResponse {
+	return &model.CommandResponse{
+		ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		Text:         fmt.Sprintf(format, args...),
+		Type:         model.POST_DEFAULT,
+	}
+}
+
+func parseLinkRef(p *Plugin, requireUnique bool, args ...string) ([]Link, []int, error) {
+	links := p.getConfig().Sorted().Links
+	if len(args) == 0 {
+		if requireUnique {
+			return nil, nil, errors.New("Unreachable")
+		}
+
+		return links, nil, nil
+	}
+
+	n, err := strconv.ParseUint(args[0], 10, 32)
+	if err == nil {
+		if n < 1 || int(n) > len(links) {
+			return nil, nil, errors.Errorf("%v is not a valid link number.", n)
+		}
+		return links, []int{int(n) - 1}, nil
+	}
+
+	found := []int{}
+	for i, l := range links {
+		if strings.Contains(l.Name, args[0]) {
+			found = append(found, i)
+		}
+	}
+	if len(found) == 0 {
+		return nil, nil, errors.Errorf("%q not found.", args[0])
+	}
+	if requireUnique && len(found) > 1 {
+		names := []string{}
+		for _, i := range found {
+			names = append(names, links[i].Name)
+		}
+		return nil, nil, errors.Errorf("%q matched more than one link: %q", args[0], names)
+	}
+
+	return links, found, nil
+}
+
+func parseBoolArg(arg string) (bool, error) {
+	switch strings.ToLower(arg) {
+	case "true", "on":
+		return true, nil
+	case "false", "off":
+		return false, nil
+	}
+	return false, errors.Errorf("Not a bool, %q", arg)
+}
+
+func saveConfigLinks(p *Plugin, links []Link) error {
+	conf := p.getConfig()
+	conf.Links = links
+	appErr := p.API.SavePluginConfig(conf.ToConfig())
```

Comment:
`.ToConfig()` here converts the `Link` structs into a tree of `map[string]interface{}` and `[]interface{}` for GOB/RPC compatibility.